### PR TITLE
Joindin 700

### PR DESCRIPTION
**Pull request by exussum12**
From 

image

To

image

First commit adds tests, second commit refactors. Split so both can be looked at, tests pass on both commits

### CODE REVIEW

File: src/controllers/BaseTalkController.php [hunk 1]
```diff
@@ -65,19 +77,6 @@ public function getUserMapper($db, $request)
         return $this->user_mapper;
     }
 
-    public function setTalkCommentMapper(TalkCommentMapper $talk_comment_mapper)
-    {
-        $this->talk_comment_mapper = $talk_comment_mapper;
-    }
-
-    public function getTalkCommentMapper($db, $request)
```
Comment (exussum12) [inline src/controllers/BaseTalkController.php]: Going to get rid of most of these functions, Lots of copy/paste to make them

File: tests/controllers/TalkLinkControllerTest.php [hunk 2]
```diff
@@ -3,12 +3,7 @@
 namespace JoindinTest\Controller;
 
 use JoindinTest\Inc\mockPDO;
-use PDOStatement;
```
Comment (exussum12) [inline tests/controllers/TalkLinkControllerTest.php]: Just clean up, these dont appear to be used

**Comment by exussum12** [comment]
Removed all the ifs by adding more routes, Should work as before

**Comment by liam-wiltshire** [comment]
ok to test

**Pull request merged**